indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceInputSplit.java

```diff
@@ -0,0 +1,64 @@
+package io.druid.indexer.hadoop;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputSplit;
+
+public class DatasourceInputSplit extends InputSplit implements Writable
+{
+  //absolute hdfs path all the way to index.zip
+  private String path;
+
+  public DatasourceInputSplit() {}
+
+  public DatasourceInputSplit(String path)
+  {
+    this.path = path;
+  }
+
+  public String getPath()
+  {
+    return path;
+  }
+
+  @Override
+  public long getLength() throws IOException
+  {
+    //TODO: may be provide a meaningful value. for now returning large enough value
+    //so that pig loader does not try to combine the splits
+    return Long.MAX_VALUE;
+  }
+
+  @Override
+  public String[] getLocations() throws IOException
+  {
+    //TODO: this is supposed to be the address of "preferred" hadoop cluster nodes
```

Comment:
`org.apache.hadoop.mapreduce.lib.input.FileSplit#getLocations` returns empty String array for unknown optimal hosts. We should do the same.